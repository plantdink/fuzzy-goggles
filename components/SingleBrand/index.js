export * from './SingleBrand';
export { default } from './SingleBrand';
