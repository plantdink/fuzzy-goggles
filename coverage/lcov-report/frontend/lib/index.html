
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for frontend/lib</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> frontend/lib</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.61% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>55/65</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">64% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>16/25</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">71.42% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>20/28</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.78% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>39/46</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="abbreviationsDictionary.js"><a href="abbreviationsDictionary.js.html">abbreviationsDictionary.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="displayNumbers.js"><a href="displayNumbers.js.html">displayNumbers.js</a></td>
	<td data-value="90.9" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="11" class="abs high">10/11</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file high" data-value="displayStrings.js"><a href="displayStrings.js.html">displayStrings.js</a></td>
	<td data-value="84.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 84%"></div><div class="cover-empty" style="width: 16%"></div></div>
	</td>
	<td data-value="84.61" class="pct high">84.61%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="testUtils.js"><a href="testUtils.js.html">testUtils.js</a></td>
	<td data-value="74.07" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 74%"></div><div class="cover-empty" style="width: 26%"></div></div>
	</td>
	<td data-value="74.07" class="pct medium">74.07%</td>
	<td data-value="27" class="abs medium">20/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="68.75" class="pct medium">68.75%</td>
	<td data-value="16" class="abs medium">11/16</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Apr 05 2022 07:32:49 GMT+1000 (Australian Eastern Standard Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    