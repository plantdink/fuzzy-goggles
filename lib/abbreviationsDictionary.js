export function deAbbreviate(abbreviation) {
  let fullTerm = '';
  if (abbreviation === 'OOC') fullTerm = 'Damage to Targets out of Cover';
  if (abbreviation === 'DTH') fullTerm = 'Damage to Health';
  if (abbreviation === 'DTA') fullTerm = 'Damage to Armour';
  if (abbreviation === 'CHC') fullTerm = 'Critical Hit Chance';
  if (abbreviation === 'CHD') fullTerm = 'Critical Hit Damage';
  if (abbreviation === 'HSD') fullTerm = 'Head Shot Damage';
  return fullTerm;
}
