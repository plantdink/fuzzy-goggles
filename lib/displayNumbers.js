export function displayPercentage(number = 0) {
  const displayNumber = number / 10;
  return displayNumber;
}

export function hsDisplay(number = 0) {
  const displayNumber = number / 100;
  return displayNumber;
}

export function millisecondsToSeconds(number = 0) {
  const displayNumber = ((number % 60000) / 1000).toFixed(2);
  return displayNumber;
}

export function humanReadableNumber(num = 0, lang = null) {
  if (!num) return;
  const locale = lang || 'en'; // removed || document.documentElement.lang
  const number = parseFloat(num, 10);
  return number.toLocaleString(locale);
}
