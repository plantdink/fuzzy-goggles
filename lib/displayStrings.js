export function lowerCaseFirstLetter(string) {
  const lowerCaseString = string.toLowerCase();
  const titleString = lowerCaseString.replace(
    /(^\w{1})|(\s+\w{1})/g,
    (letter) => letter.toUpperCase()
  );
  const noWhiteSpaceString = titleString.replaceAll(/\s/g, '');
  const finalLink =
    noWhiteSpaceString.charAt(0).toLowerCase() + noWhiteSpaceString.slice(1);
  return finalLink;
}

export function titleCase(string) {
  return string
    .toLowerCase()
    .split(' ')
    .map(function (word) {
      return word.charAt(0).toUpperCase() + word.slice(1);
    })
    .join(' ');
}

export function stringToParagraphs(string) {
  const sentences = string.split(/(?<=\.)/g);
  return sentences.map((sentence, index) => <p key={index}>{sentence}</p>);
}

export function lowerCaseLink(string) {
  const stringToEdit = string[0].toLowerCase() + string.slice(1) || string;
  return stringToEdit;
}
