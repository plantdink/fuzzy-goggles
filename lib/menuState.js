import { createContext, useContext, useState } from 'react';

const LocalStateContext = createContext();
const LocalStateProvider = LocalStateContext.Provider;

function MenuStateProvider({ children }) {
  const [menuOpen, setMenuOpen] = useState(false);

  function toggleMenu() {
    setMenuOpen(!menuOpen);
  }

  function closeMenu() {
    setMenuOpen(false);
  }

  function openMenu() {
    setMenuOpen(true);
  }

  return (
    <LocalStateProvider
      value={{
        menuOpen,
        setMenuOpen,
        toggleMenu,
        closeMenu,
        openMenu,
      }}
    >
      {children}
    </LocalStateProvider>
  );
}

// custom hook for accessing menu local state
function useMenu() {
  const all = useContext(LocalStateContext);
  return all;
}
export { MenuStateProvider, useMenu };
