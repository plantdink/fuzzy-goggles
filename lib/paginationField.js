import { WEAPON_PAGINATION_QUERY } from '../queries/PaginationQueries';

export default function paginationField() {
  // apollo will ask read for items.
  // It will return them from the cache if they exist or
  // it will return false & trigger a network request
  return {
    keyArgs: false, // toggle for apollo to let this function handle things
    read(existing = [], { args, cache }) {
      const { skip, first } = args;

      // read the number of items on the page from the cache
      const data = cache.readQuery({ query: WEAPON_PAGINATION_QUERY });
      const count = data?._allWeaponsMeta?.count;
      const page = skip / first + 1;
      const pages = Math.ceil(count / first);

      // check for existing items
      const items = existing.slice(skip, skip + first).filter((x) => x);

      // if there are items & there are not enough for the requested amount & it is the last page, go to the network
      if (items.length && items.length !== first && page === pages) {
        return items;
      }
      // there are no items, request from network
      if (items.length !== first) {
        return false;
      }

      // if there are items in cache, return them
      if (items.length) {
        return items;
      }

      // fallback to network
      return false;
    },

    // apollo will run this when the network request returns with items
    merge(existing, incoming, { args }) {
      const { skip, first } = args;
      const merged = existing ? existing.slice(0) : [];
      for (let i = skip; i < skip + incoming.length; ++i) {
        merged[i] = incoming[i - skip];
      }

      // return the merged items from the cache
      return merged;
    },
  };
}
