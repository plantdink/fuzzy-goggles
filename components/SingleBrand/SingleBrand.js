import { gql } from '@apollo/client';
import { stringToParagraphs } from '../../lib/displayStrings';

export const SINGLE_BRAND_QUERY = gql`
  query SINGLE_BRAND_QUERY($id: ID!) {
    allBrands(where: { id: $id }) {
      id
      name
      coreAttribute
      coreAttributeValueLevel40
      coreAttributeValueWT5
      setBonusOne
      setBonusTwo
      setBonusThree
      image {
        id
        altText
        image {
          publicUrlTransformed
        }
      }
      notes
      armourTalent {
        id
        name
        piece
        descriptionPVE
        isNamed
        isExotic
        image {
          id
          altText
          image {
            publicUrlTransformed
          }
        }
        namedArmourPiece {
          id
          name
        }
      }
      armourType {
        id
        name
        piece
        isNamed
        image {
          id
          altText
          image {
            publicUrlTransformed
          }
        }
      }
    }
  }
`;

export default function SingleBrand({ brand }) {
  return (
    <>
      <div className="single-gear-item__content" data-testid="testBrand">
        <div className="single-gear-item__details">
          {stringToParagraphs(brand.notes)}
          <table>
            <thead className="single-gear-item__title-bar">
              <tr>
                <th className="single-gear-item__subheading">
                  Core Attribute Type - {brand.coreAttribute}
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th scope="row">Level 40</th>
                <td>+ {brand.coreAttributeValueLevel40}</td>
              </tr>
              <tr>
                <th scope="row">World Tier 5</th>
                <td>+ {brand.coreAttributeValueWT5}</td>
              </tr>
            </tbody>
          </table>

          <table>
            <thead className="single-gear-item__title-bar">
              <tr>
                <th className="single-gear-item__subheading">Set Bonuses</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>Whilst wearing:</td>
              </tr>
              <tr>
                <th scope="row" className="single-gear-item__sub-subheading">
                  1 Piece
                </th>
                <td>{brand.setBonusOne}</td>
              </tr>
              <tr>
                <th scope="row">2 Piece</th>
                <td>{brand.setBonusTwo}</td>
              </tr>
              <tr>
                <th scope="row">3 Piece</th>
                <td>{brand.setBonusThree}</td>
              </tr>
            </tbody>
          </table>
        </div>

        <div className="item-image" data-testid="testImage">
          <img
            className="standard-item"
            id="brand-page__image"
            alt={brand.image?.altText}
            src={brand.image?.image.publicUrlTransformed}
          />
        </div>
      </div>
    </>
  );
}
