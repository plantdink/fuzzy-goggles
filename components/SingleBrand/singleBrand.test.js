import { render, screen, cleanup } from '@testing-library/react';
import SingleBrand from '.';
import { fakeBrand } from '../../lib/testUtils';

const brand = fakeBrand();

afterEach(cleanup);

describe('<SingleBrand />', () => {
  it('renders correctly for a single brand', async () => {
    const { container, debug } = render(<SingleBrand brand={brand} />);

    await screen.findByTestId('testBrand');
    expect(container).toMatchSnapshot();

    const testBrandCoreAttribute = screen.getByRole('row', {
      name: /core attribute type \- armour/i,
    });
    expect(testBrandCoreAttribute).toHaveTextContent(brand.coreAttribute);

    const testCoreAttribute40Value = screen.getByRole('cell', {
      name: /\+ 170,000 armour/i,
    });
    expect(testCoreAttribute40Value).toHaveTextContent(
      brand.coreAttributeValueLevel40
    );

    const testCoreAttributeWT5Value = screen.getByRole('cell', {
      name: /\+ 42,458 armour/i,
    });
    expect(testCoreAttributeWT5Value).toHaveTextContent(
      brand.coreAttributeValueWT5
    );

    const testSetBonusOne = screen.getByRole('row', { name: /1 piece/i });
    expect(testSetBonusOne).toHaveTextContent(brand.setBonusOne);

    const testSetBonusTwo = screen.getByRole('row', { name: /2 piece/i });
    expect(testSetBonusTwo).toBeInTheDocument(brand.setBonusTwo);

    const testSetBonusThree = screen.getByRole('row', {
      name: /3 piece/i,
    });
    expect(testSetBonusThree).toBeInTheDocument(brand.setBonusThree);

    const testBrandImage = screen.getByRole('img', {
      name: /fake brand image alt text/i,
    });
    expect(testBrandImage).toBeInTheDocument();
    expect(testBrandImage).toHaveAttribute(
      'src',
      `${brand.image.image.publicUrlTransformed}`
    );
    expect(testBrandImage).toHaveAttribute('alt', `${brand.image.altText}`);
  });
});
