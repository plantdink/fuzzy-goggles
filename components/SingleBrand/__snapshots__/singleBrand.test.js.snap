// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleBrand /> renders correctly for a single brand 1`] = `
<div>
  <div
    class="single-gear-item__content"
    data-testid="testBrand"
  >
    <div
      class="single-gear-item__details"
    >
      <p>
        Strategic Homeland Division has approved multiple Brand manufacturers to supply equipment for SHD agents.
      </p>
      <p>
         There are also numerous non-approved Brands that an agent may find during the conduct of operations and agents are authorised to use any equipment they need in order to complete their mission.
      </p>
      <p>
         Each Brand provides different benefits to an agent and their ability to conduct operations.
      </p>
      <p>
         Equipping items from the same brand will unlock different benefits, with 3 equipped providing the maximum Brand-specific benefit.
      </p>
      <p>
         Agents are encouraged to mix and match Brands to provide the best benefit to their preferred operational style.
      </p>
      <table>
        <thead
          class="single-gear-item__title-bar"
        >
          <tr>
            <th
              class="single-gear-item__subheading"
            >
              Core Attribute Type - 
              Armour
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              scope="row"
            >
              Level 40
            </th>
            <td>
              + 
              170,000 Armour
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              World Tier 5
            </th>
            <td>
              + 
              42,458 Armour
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead
          class="single-gear-item__title-bar"
        >
          <tr>
            <th
              class="single-gear-item__subheading"
            >
              Set Bonuses
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Whilst wearing:
            </td>
          </tr>
          <tr>
            <th
              class="single-gear-item__sub-subheading"
              scope="row"
            >
              1 Piece
            </th>
            <td>
              10% Health
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              2 Piece
            </th>
            <td>
              20% Extra Incoming Healing
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              3 Piece
            </th>
            <td>
              10% Hazard Protection
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="item-image"
      data-testid="testImage"
    >
      <img
        alt="Fake Brand image alt text"
        class="standard-item"
        id="brand-page__image"
        src="FakeNamedBrand.jpg"
      />
    </div>
  </div>
</div>
`;
